### DISCUSSION TITLE
Opening an Event automatically opens the first Text Box

### DISCUSSION THREAD

**Discussion by MercurioC** [4ian/GDevelop#3227] [Enhancement ideas (and their technical discussions)]
## Describe the bug
As the title states, double-clicking on any event (no matter if it's a Condition or an Action) causes the program to immediately open up the first text box available. Nothing major but kinda annoying in the long run :-)

## To Reproduce
Just double-click on any created Event in the Events page. Here's a gif showing it:
[Imgur](https://i.imgur.com/nwI7Hub.gifv)

## Other details
* Win10 17134, both on desktop and the web-app with Chrome 69.0.3497.100

**Comment by blurymind** [comment]
I see  what you mean. The text box immediately steals focus and pops up with the auto completion list- which is a bit distracting, because the autocompletion overlaps over some of the information of the event
![nwi7hub](https://user-images.githubusercontent.com/6495061/46233624-487f0100-c36b-11e8-9ca3-3bdb03a67051.gif)

I think this is easy to fix, but first I want to see if this is intended design by @4ian . Perhaps it helps productivity by anticipating what the user might want?

**Comment by blurymind** [comment]
Do you think that the value input should steal focus instead? Or maybe nothing should steal focus if the user is editing? The search steals focus when creating - which makes a lot of sense

**Comment by 4ian** [comment]
It is intended design (search box get focused when window opened for a new condition/action or first parameter for an existing one), but I agree that having the menu popping up is a bit of a pain.
There is maybe a may to keep the focus but disable the menu by removing `openOnFocus` on the `ObjectSelector`.

**Comment by blurymind** [comment]
that would resolve it imo. The unintended popup is the only issue here.
Could give it a try tomorow if you havent fixed it already :p there are so
many interesting things to pick up in gd atm

**Comment by 4ian** [comment]
I'm at a conference to speak about GDevelop (http://www.reactboston.com/) tomorrow and working on other things so feel free to pick this :)

**Comment by MercurioC** [comment]
Disabling the menu but keeping the focus on the box seems the way to go, I agree 👍 

Thanks for all your work both of you BTW! This is the first time I've opened an issue for anything on Git, I've discovered GDevelop just recently by pure luck and it's really what I was looking for during all these years grasping with GameMaker, MMF2 and whatnot. Keep it going! :-)

**Comment by blurymind** [comment]
Playing with this for a bit, I find that I like the design of having autocomplete pop up when an input element is focused. The problem is in the element triggering it when stealing focus automatically in this specific case.

So a simple solution to this is to just not steal focus in this case.
You want to evaluate what is in the event before doing anything with it. You would also want that autocomplete to be there when wanting to change that value. Not having it there, but being in focus can create some confusion

**Comment by blurymind** [comment]
submitted a simple fix for this
:)

**Comment by 4ian** [comment]
The thing is that it was done for enabling modifying the parameters with keyboard without using the mouse at all. Can we find another solution then?

**Comment by blurymind** [comment]
@4ian maybe the autocomplete could pop up upon click instead of onfocus?
But with a keyboard only interaction you would still want to have the autocomplete.  

Maybe somehow disable autocomplete pop up from appearing in some specific corner cases such as this? And when the user types with keyboard, it appears?

**Comment by 4ian** [comment]
Yes ideally it would appear after the first interaction when it has the focus. Not sure how easy it is to achieve. Might be good to play a bit with the Autocomplete component from Material UI: https://v0.material-ui.com/#/components/auto-complete

**Comment by Silver-Streak** [comment]
@4ian @blurymind 

I'm trying to clean up old threads, and this came up on my list. This is still an ongoing behavior with the event list in b114. Is the intent to still change this, or should I close this issue out?

**Comment by Bouh** [comment]
It would be nice to keep it open.
The autofocus that opens the suggestions is not good, I agree we have to put a click as he said above.
